Code/Source/Lidar/LidarRaycaster.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "LidarRaycaster.h"
+#include <AzFramework/Physics/PhysicsScene.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzFramework/Physics/Common/PhysicsSceneQueries.h>
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+namespace ROS2
+{
+    // A simplified, non-optimized first version. TODO - generalize results (fields)
+    AZStd::vector<AZ::Vector3> LidarRaycaster::PerformRaycast(const AZ::Vector3& start, const AZStd::vector<AZ::Vector3>& directions, float distance)
+    {
+        AZStd::vector<AZ::Vector3> results;
+        AzPhysics::SceneQueryRequests requests;
+        for (const AZ::Vector3& direction : directions)
+        {   // NOTE - performance-wise, consider reusing requests
+            AZStd::shared_ptr<AzPhysics::RayCastRequest> request = AZStd::make_shared<AzPhysics::RayCastRequest>();
+            request->m_start = start;
+            request->m_direction = direction;
+            request->m_distance = distance;
+            request->m_reportMultipleHits = false;
+            requests.emplace_back(AZStd::move(request));
+        }
```

Comment:
This could `reserve` the size of the `directions` vector for the `requests` vector to reduce the number of allocations.